ci/build_container/build_container.sh

```diff
@@ -46,3 +46,10 @@ export THIRDPARTY_BUILD=/thirdparty_build
 export CPPFLAGS="-DNDEBUG"
 echo "Building opt deps ${DEPS}"
 "$(dirname "$0")"/build_and_install_deps.sh ${DEPS}
+
+echo "Building Bazel-managed deps (//bazel/external:all_external)"
+BAZEL_OPTIONS="--output_user_root=/bazel-prebuilt/BAZEL_ROOT --output_base=/bazel-prebuilt/BAZEL_OUTPUT"
+cd /bazel-prebuilt
+for BAZEL_MODE in opt dbg fastbuild; do
+  bazel ${BAZEL_OPTIONS} build -c "${BAZEL_MODE}" //bazel/external:all_external
```

Comment:
I'm good with this approach, providing we maintain the `bind` support so that folks who really want prebuilts (e.g. that don't come from our build process), are capable of arbitrary rebinding in their `WORKSPACE`.

Reply:
Rather than using `bind()`, we should encourage users to override deps using Bazel repositories. That's why the `existing_rule_keys` stuff is used -- a user can define a different `@protobuf_bzl` repository in their own `WORKSPACE`.

See comments on https://docs.bazel.build/versions/master/be/workspace.html#bind and discussion on https://github.com/bazelbuild/bazel/issues/1952 ("Consider removing bind()") for more background on discouraging use of `bind()`.

Reply:
As long as we can support remapping of the external repos via `envoy_exernal_dep_path`, we're OK with this at Google. Internally, as you know, we don't have any `@` repos, so we need to be able to take each of the individual repository/dependency labels and remap them to a global path in our monorepo.

Reply:
What is `envoy_exernal_dep_path`? Is that the same as the `path` param to `envoy_dependencies`?

How do you currently handle the `bind()` names? I assume there's something set up to avoid polluting the namespace with things like `//external:zlib` -- do those get transparently mapped to `//third_party/` somehow?

Reply:
https://github.com/envoyproxy/envoy/blob/master/bazel/envoy_build_system.bzl#L71. Internally, we rewrite the entire `envoy_build_system.bzl` to match Google's monorepo and build requirements. That function gets turned into a dictionary lookup, that maps from identifiers such as `ares` to something in `//third_party`.

Reply:
Ah, that should be fine then. This PR does make a small change to `envoy_external_dep_path` to support non-ambiguous dep names, but that's merely a change in the identifiers. And since none of them are changing in this PR, you don't have to do anything.

Reply:
What's the Bazel story for aliasing without `bind()`? Suppose I have a repository under `@foo`, I import `@bar`, and it expects `@foo` to exists as `@baz`. How do I tell Bazel to find `@baz` as `@foo`?